src/pages/index.js

```diff
@@ -30,17 +34,69 @@ const enableValidation = (settings) => {
 };
 
 function handleImageFormSubmit(name, link) {
-  renderCard({ name: name, link: link });
+  addImagePopup.renderLoading(true);
+  api
+    .addNewCard({ name, link })
+    .then((newCardData) => {
+      const { name, link } = newCardData;
+      cardSection.addItem(
+        new Card(
+          newCardData,
+          "#card-template",
+          imagePopup,
+          (title, link) => imagePopup.open(title, link),
+          (cardElement, cardID) => {
+            deleteImagePopup.open(cardElement, newCardData._id);
+          },
+          api
+        ).getView()
+      );
+    })
+    .catch((err) => console.error("Error adding new card:", err));
   addImagePopup.close();
 }
 
 function handleProfileFormSubmit(name, description) {
-  userInfo.setUserInfo({
-    name: name,
-    job: description,
-  });
+  editProfilePopup.renderLoading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```

Reply:
Implemented. Took me a second to understand, but I like it. Thanks for the suggestion!